From 8f042b1af8816d9e40827c97fa610433545802e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Mon, 6 Sep 2021 00:44:56 +0200
Subject: [PATCH] 0032328: Missing include of TopoDS_Edge.hxx in
 ShapeUpgrade_UnifySameDomain.hxx

Commit 91e51cb0f2ec0b59 ("0032140: Modeling Algorithms - unify same
domain calls crossed for opposite vectors") moved the definition of
SubSequenceOfEdges to the header files.

As it is used only via reference in the header file a forward declaration
is sufficient. Move the definition of SubSequenceOfEdges back to the
implementation file.

This fixes any compilation failures due to the missing definition of
TopoDS_Edge.
---
 src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.cxx | 6 ++++++
 src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.hxx | 7 +------
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.cxx b/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.cxx
index da8e4f16be..64d86c7681 100644
--- a/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.cxx
+++ b/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.cxx
@@ -105,6 +105,12 @@ static void SplitWire (const TopoDS_Wire&                theWire,
                        const TopTools_IndexedMapOfShape& theVmap,
                        TopTools_SequenceOfShape&         theWireSeq);
 
+struct SubSequenceOfEdges
+{
+  TopTools_SequenceOfShape SeqsEdges;
+  TopoDS_Edge UnionEdges;
+};
+
 static Standard_Real TrueValueOfOffset(const Standard_Real theValue,
                                        const Standard_Real thePeriod)
 {
diff --git a/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.hxx b/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.hxx
index b1558d111f..9c0379148e 100644
--- a/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.hxx
+++ b/src/ShapeUpgrade/ShapeUpgrade_UnifySameDomain.hxx
@@ -32,6 +32,7 @@
 #include <Precision.hxx>
 class ShapeBuild_ReShape;
 class TopoDS_Shape;
+struct SubSequenceOfEdges;
 
 
 class ShapeUpgrade_UnifySameDomain;
@@ -66,12 +67,6 @@ DEFINE_STANDARD_HANDLE(ShapeUpgrade_UnifySameDomain, Standard_Transient)
 //! history by default.
 //! To avoid collecting of the history the place holder should be set to null handle.
 
-struct SubSequenceOfEdges
-{
-  TopTools_SequenceOfShape SeqsEdges;
-  TopoDS_Edge UnionEdges;
-};
-
 class ShapeUpgrade_UnifySameDomain : public Standard_Transient
 {
 
-- 
2.32.0

